dali/test/python/test_operator_peek_shape.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import nvidia.dali.fn as fn
+import nvidia.dali as dali
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.types as types
+import os
+import numpy as np
+from test_utils import get_dali_extra_path
+
+test_data_root = get_dali_extra_path()
+path = 'db/single'
+file_types = {'jpeg', 'mixed', 'png', 'tiff', 'pnm', 'bmp'}
+
+dali_to_numpy = {
+    types.INT32   : np.int32,
+    types.INT64   : np.int64,
+    types.UINT32  : np.uint32,
+    types.UINT64  : np.uint64,
+    types.FLOAT   : np.float32,
+    types.FLOAT64 : np.float64,
+}
+
+def run_decode(data_path, device, out_type):
+    batch_size = 4
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=0)
+    input, _ = fn.file_reader(file_root=data_path, shard_id=0, num_shards=1, name="reader")
+    decoded = fn.image_decoder(input, device=device, output_type=types.RGB)
+    if device == "cpu" :
+        shape_device = "cpu"
+    else:
+        shape_device = "gpu"
+    decoded_shape = fn.shapes(decoded, device=shape_device)
+    raw_shape = fn.peek_shape(input, type=out_type)
+    pipe.set_outputs(decoded, decoded_shape, raw_shape)
+    pipe.build()
+    samples = 0
+    length = pipe.reader_meta(name="reader")['epoch_size']
+    while samples < length:
+        samples += batch_size
+        out = pipe.run()
+        (image, decoded_shape, raw_shape) = [v.as_cpu() if isinstance(v, dali.backend_impl.TensorListGPU) else v for v in out]
+        for i in range(batch_size):
+            # as we are asking for a particular color space it may differ from the source image, so don't comapre it
+            for d in range(len(image.at(i).shape) - 1):
+                shape_type = dali_to_numpy[out_type]
+                assert image.at(i).shape[d] == decoded_shape.at(i)[d], "{} vs {}".format(image.at(i).shape[d], decoded_shape.at(i)[d])
+                assert image.at(i).shape[d] == raw_shape.at(i)[d], "{} vs {}".format(image.at(i).shape[d], raw_shape.at(i)[d])
+                assert raw_shape.at(i)[d].dtype == shape_type, "{} vs {}".format(raw_shape.at(i)[d].dtyp, shape_type)
+
+test_types = [types.INT32, types.UINT32, types.INT64, types.UINT64, types.FLOAT, types.FLOAT64]
+
+def test_image_decoder():
+    for device in {'cpu', 'mixed'}:
```

Comment:
```suggestion
    for device in {'cpu', 'gpu'}:
```
I'd rather set the device of the PeekShape operator here and calculate the device for decoder inside the function

Reply:
Peek shape is available only for the CPU, so it doesn't make much sense. Maybe I should just test for the CPU, after all.

Reply:
Done